src/blenderbim/blenderbim/bim/import_ifc.py

```diff
@@ -684,6 +686,62 @@ def create_structural_point_connections(self):
             obj.matrix_world = self.apply_blender_offset_to_matrix_world(obj, placement_matrix)
             self.link_element(product, obj)
 
+    def create_pointclouds(self, products):
+        result = set()
+        for product in products:
+            if not self.is_pointcloud(product):
+                continue
+            items = []
+            for representation in product.Representation.Representations:
+                if representation.RepresentationType == 'PointCloud':
+                    items.extend(representation.Items)
+                elif self.file.schema == "IFC2X3" and representation.RepresentationType == 'GeometricSet':
+                    items.extend(representation.Items)
+
+            placement_matrix = ifcopenshell.util.placement.get_local_placement(product.ObjectPlacement)
+            vertex_list = []
+            context = None
+            representation = None
+            for item in items:
+                if item.is_a("IfcCartesianPointList"):
+                    vertices = [mathutils.Vector(list(coordinates)) * self.unit_scale for coordinates in item.CoordList]
+                    vertex_list.extend(vertices)
+                elif item.is_a("IfcCartesianPoint"):
+                    vertex = mathutils.Vector(list(item.Coordinates)) * self.unit_scale
+                    vertex_list.append(vertex)
+
+            for subelement in self.file.traverse(product.Representation):
+                if subelement.is_a("IfcGeometricRepresentationContext"):
+                    context = subelement
+                elif subelement.is_a("IfcRepresentation"):  # not sure what this should be
+                    representation = subelement
+
+            if len(vertex_list) == 0 or not context or not representation:
+                continue
+
+            mesh_name = f"{context.id()}/{representation.id()}"
+            mesh = bpy.data.meshes.new(mesh_name)
+            mesh.from_pydata(vertex_list, [], [])
+
+            obj = bpy.data.objects.new("{}/{}".format(product.is_a(), product.Name), mesh)
+            obj.matrix_world = self.apply_blender_offset_to_matrix_world(obj, placement_matrix)
+            self.link_element(product, obj)
+            result.add(product)
+
+        return result
+
+    def is_pointcloud(self, product):
+        if not hasattr(product, 'Representation') or not hasattr(product.Representation, 'Representations'):
+            return False
+
+        for representation in product.Representation.Representations:
+            if representation.RepresentationType == 'PointCloud':
+                return True
+            elif self.file.schema == "IFC2X3" and representation.RepresentationType == 'GeometricSet':
```

Comment:
In 2X3 we may need to check the type of the items to ensure that it is storing points, as opposed to curves / surfaces.